joomla/joomla-cms
# [3.9] Fixed left-right operator precedence in showon expressions.

Pull Request for Issue #16954 .

### Summary of Changes
The operator precedence in showon expressions was not working as expected, that is, from left to right.  All operators prior to the right-most were being ignored completely.

### Testing Instructions
Set up a form using the XML file (which is actually the .txt file in the attached zip).  This has three fields (Field 1, Field 2 and Field3) which can each be set to Yes or No.  Depending on the state of these fields, further fields will be shown or hidden according to the showon expressions they contain.  All possible operator combinations for the three fields are included.  Change the field states and compare with the spreadsheet in the attached zip which shows the actual and expected results.

### Expected result
As per the Expected section of the spreadsheet in the attached zip.

### Actual result
As per the Actual section of the spreadsheet in the attached zip.

### Documentation Changes Required
Add a note to https://docs.joomla.org/Form_field to explain that left-right precedence should be expected.


[showon-issue-16954.zip](https://github.com/joomla/joomla-cms/files/2971796/showon-issue-16954.zip)
